src/main/scala/com/epam/reportportal/scalatest/RPReporter.scala

```diff
@@ -0,0 +1,138 @@
+      override def get() = Injector.getInstance.getChildInjector(new TestContextProvider).getBean(classOf[ReporterServiceImp])
+    })
+  }
+
+  /*
+   * The reportportal.properties is not loaded by client-java-core using sbt goal, we load it into System properties here.
+   * It is a workaround.
+   */
+  private[scalatest] def loadReportPortalProperties(): Unit = {
+    try {
+      val properties: Properties = new Properties()
+      val inputStream: InputStream = getClass().getClassLoader().getResourceAsStream(reportPortalPropertiesFileName)
+      properties.load(inputStream)
+      properties.stringPropertyNames().toArray().foreach(p =>
+        System.setProperty(p.toString, properties.getProperty(p.toString)))
+    } catch {
+      case e: Exception => logger.warn(s"$reportPortalPropertiesFileName file is not found", e)
+    }
+  }
+
+  def apply(event: Event): Unit = event match {
+
+    case e: RunStarting ⇒ {
+
+      reporterService.get().startLaunch(e)
+    }
+
+    case e: RunCompleted ⇒ {
+      reporterService.get().finishLaunch(e)
+    }
+
+    case e: SuiteStarting ⇒ {
+      val klass = Class.forName(e.suiteClassName.get)
+      // we don't report yet suits that has nested suites
+      if (Class.forName("org.scalatest.Suites").isAssignableFrom(klass)) {
+        // will be implemented later
+        // Do nothing right now
+      } else {
+        reporterService.get().startTestClass(e)
+      }
+    }
+
+    case e: SuiteCompleted ⇒ {
+      val klass = Class.forName(e.suiteClassName.get)
+      // we don't report yet suits that has nested suites
+      if (Class.forName("org.scalatest.Suites").isAssignableFrom(klass)) {
+        // will be implemented later
+        // Do nothing right now
+      } else {
+        reporterService.get() finishTestClass (e: SuiteCompleted)
+      }
+    }
+
+    case (e: SuiteAborted) ⇒ {
+      val klass = Class.forName(e.suiteClassName.get)
+      // we don't report yet suits that has nested suites
+      if (Class.forName("org.scalatest.Suites").isAssignableFrom(klass)) {
+        // will be implemented later
+        // Do nothing right now
+      } else {
+        reporterService.get() finishTestClass (e)
+      }
+    }
+
+    case e: TestStarting ⇒ {
+      reporterService.get().startTestMethod(e)
+    }
+
+    case e: TestCanceled ⇒ {
+      reporterService.get().finishTestMethod(e)
+    }
+
+    case e: TestFailed ⇒ {
+      reporterService.get().sendReportPortalMsg(e)
+      reporterService.get().finishTestMethod(e)
+    }
+
+    case e: TestIgnored ⇒ {
+      reporterService.get().startTestMethod(e)
+      reporterService.get().finishTestMethod(e)
+    }
+
+    case e: TestPending ⇒ {
+      reporterService.get().finishTestMethod(e)
+    }
+
+    case e: TestSucceeded ⇒ {
+      reporterService.get().finishTestMethod(e)
+    }
+
+    case e: InfoProvided ⇒ {
+      println(e.getClass.getSimpleName)
+    }
+
+    case e: DiscoveryStarting ⇒ println(e.getClass.getSimpleName)
```

Comment:
it would be better to use some logger here. The best option is to use slf4j as logging facade